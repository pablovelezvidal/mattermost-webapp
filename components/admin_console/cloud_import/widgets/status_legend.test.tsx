// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import {mountWithIntl} from 'tests/helpers/intl-test-helper';

import StatusLegend from './status_legend';

describe('admin_console/status_legend', () => {
    test('status legend match snapshot', () => {
        const wrapper = shallow(
            <StatusLegend status='completed'/>,
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('status legend shows failed', () => {
        const wrapper = mountWithIntl(
            <StatusLegend status='failed'/>,
        );
        expect(wrapper).toMatchSnapshot();
        const span = wrapper.find('span');
        expect(span.text()).toBe('Failed');
    });

    test('status legend shows completed', () => {
        const wrapper = mountWithIntl(
            <StatusLegend status='completed'/>,
        );
        expect(wrapper).toMatchSnapshot();
        const span = wrapper.find('span');
        expect(span.text()).toBe('Completed');
    });

    // commented out since the in progress icon does not exist yet and using img + gif breaks the snapshot
    // test('status legend shows in progress', () => {
    //     const wrapper = mountWithIntl(
    //         <StatusLegend status='in_progress'/>,
    //     );
    //     expect(wrapper).toMatchSnapshot();
    //     const span = wrapper.find('span');
    //     expect(span.text()).toBe('In Progress');
    // });
});
