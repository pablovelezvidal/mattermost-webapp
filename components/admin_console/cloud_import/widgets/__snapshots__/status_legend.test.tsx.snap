// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`admin_console/status_legend status legend match snapshot 1`] = `
<div
  className="StatusLegend completed"
>
  <i
    className="icon icon-check-circle-outline"
  />
  <FormattedMessage
    defaultMessage="Completed"
    id="admin.general.completed"
  />
</div>
`;

exports[`admin_console/status_legend status legend shows completed 1`] = `
<StatusLegend
  status="completed"
>
  <div
    className="StatusLegend completed"
  >
    <i
      className="icon icon-check-circle-outline"
    />
    <FormattedMessage
      defaultMessage="Completed"
      id="admin.general.completed"
    >
      <span>
        Completed
      </span>
    </FormattedMessage>
  </div>
</StatusLegend>
`;

exports[`admin_console/status_legend status legend shows failed 1`] = `
<StatusLegend
  status="failed"
>
  <div
    className="StatusLegend failed"
  >
    <i
      className="icon icon-alert-outline"
    />
    <FormattedMessage
      defaultMessage="Failed"
      id="admin.general.failed"
    >
      <span>
        Failed
      </span>
    </FormattedMessage>
  </div>
</StatusLegend>
`;
